tests/code/test_fileHandler.py

```diff
@@ -3,68 +3,124 @@
 def test_retrievePath():
-    # test event
-    eventfiles= AttachmentUpload.select().where(AttachmentUpload.event == 15)
-    paths = handledEventFile.retrievePath(eventfiles)
-    path = paths["eventfile.pdf"][0]
-    assert path =='/static/files/eventattachments/15/eventfile.pdf'
+    with mainDB.atomic() as transaction:
+
+        # uploads attachments for course test to the database using the transaction
+        AttachmentUpload.create(course=1, fileName= 'coursefile.pdf')
+
+        # uploads attachments for events tests to the database using the transaction
+        AttachmentUpload.create(event=15, fileName= '15/eventfile.pdf')
+
+        AttachmentUpload.create(event=16, fileName= '15/eventfile.pdf')
 
-    # test course
-    coursefiles= AttachmentUpload.select().where(AttachmentUpload.course == 1)
-    paths = handledCourseFile.retrievePath(coursefiles)
-    path = paths["coursefile.pdf"][0]
-    assert path =='/static/files/courseattachments/1/coursefile.pdf'
+
+        # test event
+        eventfiles= AttachmentUpload.select().where(AttachmentUpload.event == 15)
+        paths = handledEventFile.retrievePath(eventfiles)
+        path = paths['15/eventfile.pdf'][0]
+        assert path =='/static/files/eventattachments/15/eventfile.pdf'
+
+        # test recurring event
+        # this tests that the recurring events are referencing the same file directory as the first event.
+        eventfiles= AttachmentUpload.select().where(AttachmentUpload.event == 16)
+        paths = handledEventFile.retrievePath(eventfiles)
+        path = paths['15/eventfile.pdf'][0]
+        assert path =='/static/files/eventattachments/15/eventfile.pdf'
+
+        # test course
+        coursefiles= AttachmentUpload.select().where(AttachmentUpload.course == 1)
+        paths = handledCourseFile.retrievePath(coursefiles)
+        path = paths["coursefile.pdf"][0]
+        assert path =='/static/files/courseattachments/1/coursefile.pdf'
+
+        transaction.rollback()
 
 @pytest.mark.integration
 def test_deleteFile():
-    # test file
-    eventfiles= AttachmentUpload.select().where(AttachmentUpload.event == 15)
-    pathDictionary = handledEventFile.retrievePath(eventfiles)
-    fileId = pathDictionary["eventfile.pdf"][1]
-    path = pathDictionary["eventfile.pdf"][0]
-    handledEventFile.deleteFile(fileId)
-    assert os.path.exists(path)==False
+    with mainDB.atomic() as transaction:
+        # creates file in event file directory for deletion
+        handledEventFile.saveFiles(saveOriginalFile = Event.get_by_id(15))
+
+        # creates a second file to simulate recurring events
+        handledEventFileRecurring.saveFiles(saveOriginalFile = Event.get_by_id(15))
+        
+        # creates a course file for deletion
+        handledCourseFile.saveFiles()
+
+        # delete events
+        path = handledEventFile.getFileFullPath("15/eventfile.pdf")
+
+        firstevent = AttachmentUpload.get(AttachmentUpload.event == 15)
```

Comment:
I'd put a comment before this piece of the test clarifying why the path should still exist after the `deleteFile` call